source/agora/common/Amount.d

```diff
@@ -139,6 +139,34 @@ public struct Amount
         return copy;
     }
 
+    /// Ditto
+    public Amount opBinary (string op : "*") (in ulong multiplier) const scope
+        @safe
+    {
+        Amount copy = this;
+        if (!copy.mul(multiplier))
+            throw new Exception(format("Amount %d cannot be multiplied by %d", this, multiplier));
+        return copy;
+    }
+
+    /// Return the amount divided by denominator ignoring the remainder (use % to get remainder)
+    public Amount opBinary (string op : "/") (in ulong denominator) const scope
+        @safe
+    {
+        Amount copy = this;
+        copy.div(denominator);
+        return copy;
+    }
+
+    /// Return the remainder of division
+    public Amount opBinary (string op : "%") (in ulong denominator) const scope
+        @safe
+    {
+        Amount copy = this;
+        return copy.div(denominator);
+    }
+
+
```

Comment:
The mul makes sense, but the div / modulo, where is it used ? The core tenet of `Amount` is that you can't get it wrong. The `opBinary` additions still respect this, by giving your a runtime Exception (having just `add` / `sub` meant you were forced to handle it statically). But discarding part of the `Amount` is not something we should encourage, so if there's only one or two usages of it, I'd rather not introduce the overloads.

Reply:
How about if we make them `version unittest`. The idea was to make the test code less verbose and easier to understand.

Reply:
How often are they used though ? I don't expect it to be that often ?

Reply:
I refactored to not need the `/` and `%` `opBinary` functions as we can not make it `version unittest` as it is used in other tests which are not `version unittest` like block storage test.